services/src/pro/quota/mod.rs

```diff
@@ -128,12 +128,12 @@ impl<U: UserDb + 'static> QuotaManager<U> {
 
                 if flush_buffer {
                     // TODO: what to do if this fails (quota can't be recorded)? Try again later?
-                    for (user, quota_used) in self.increment_quota_buffer.drain() {
-                        let r = self.user_db.increment_quota_used(&user, quota_used).await;
+                    let quota_updates = self.increment_quota_buffer.drain().collect::<Vec<_>>();
```

Comment:
Do you convert a HashMap from user_id -> number of chunks to an array.
Then, you split the array into two arrays and put it into the db an then unnest it?

You could stick with the map, right?
Or you could insert keys() and values() so that you don't need to copy the vec twice.

Reply:
it could certainly be simplified. But I think u64 makes more sense for the interface (trait method), as the used quota should never be negative. But: Postgres has no unsigned datatype, so we have to convert it to i64 before executing the query.

> You could stick with the map, right?

I don't think it's possible to insert a map into postgres, if that's what you mean. But if you have an idea let me know.

> Or you could insert keys() and values() so that you don't need to copy the vec twice.

I don't think it's that easy, because then we would lose the correspondence between key and value. We need to collect the entries from the hasmap I think. Splitting it up into separate Vecs could be avoided I guess but I don't really think it's worth it.

Reply:
Ok, then just change the parameter to be an Iter<(UserId, u64)>, then you don't have to create a vec and then create two new vecs.

Reply:
what do you mean with `Iter<(UserId, u64)>`? 

Do you mean `std::collections::hash_map::Iter`? Then we'd be limited to HashMaps and could pass the HashMap as it is.

Or do you mean `Iterator`? That's a trait, which means we would have to make the function generic, or pass a fat pointer.

Reply:
You can just make it `function<I: Iterator<Item=(UserId, u64)>>`.

Reply:
yeah that's what I meant. It would make the function generic, leading to more code being generated and restricting the use of this trait e.g. for usage as a trait object. Is it really worth it?

Reply:
8ac8eafc4fe835edc5611a01f9f00ac1421a6b0d